### PR TITLE
Issue #15 and improvements

### LINKED ISSUE #15: Add exception handlers to ListBox
Object and Coordinate ListBox throw errors if no row is selected.

### PR DESCRIPTION
**Pull request by ramnaq**
Closes #15 
Fixes, avoiding errors, and some improvements

### CODE REVIEW

File: src/controller.hpp [hunk 1]
```diff
@@ -176,7 +184,11 @@ class Controller {
    */
   void changeWindow(int op) {
     double step = view.getStep();
-    view.updateWindow(step, op);
+	try {
```
Comment (FabioMoreiraFM) [inline src/controller.hpp:191]: spacing problem

File: src/logger.hpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    void logError(string err) {
+	  logMessage("[ERRO] " + err);
+	}
+
+  private:
+    GtkTextView* textView;
+	GtkTextBuffer* buffer;
+
+	GtkTextIter iter;
```
Comment (FabioMoreiraFM) [inline src/logger.hpp:29]: Spacing problem

**Review by FabioMoreiraFM — CHANGES_REQUESTED** [review]
The system isn't cleaning correctly the polygon coordinate list. The first time you insert the coordinates, if you insert less than 3 coordinates, the system don't allow. But if you insert correctly one polygon and after that erase one or more coordinates, you can draw a new "polygon" with less than 3 coordinates. If you try to remove this "polygon", the system breaks.

The system isn't drawing the right figure. For instance (100, 5) (200, 5) (200, 100) (100, 100) is supposed to draw a square, but it is drawed a triangle.

Some parts of the code are with spacing problems (I pointed out with comments.)

**Comment by ramnaq** [comment]
> The system isn't cleaning correctly the polygon coordinate list. The first time you insert the coordinates, if you insert less than 3 coordinates, the system don't allow. But if you insert correctly one polygon and after that erase one or more coordinates, you can draw a new "polygon" with less than 3 coordinates. If you try to remove this "polygon", the system breaks.
> 
> The system isn't drawing the right figure. For instance (100, 5) (200, 5) (200, 100) (100, 100) is supposed to draw a square, but it is drawed a triangle.
> 
> Some parts of the code are with spacing problems (I pointed out with comments.)

About the second problem: corrected! It was a mistake: "size() - 1" not needing "-1".

**Comment by ramnaq** [comment]
Status: changes done. I'm not able to push them now. They will be in the remote branch until tomorrow's morning.

**Review by FabioMoreiraFM — APPROVED** [review]

**Pull request merged**